src/libos/src/net/socket/socket.rs

```diff
@@ -0,0 +1,14 @@
+use super::*;
+
+// Trait for Berkeley sockets API
+// TODO: add more functions
+pub trait Socket {
+    fn bind(&self, addr: &SockAddr) -> Result<()>;
+    fn listen(&self, backlog: i32) -> Result<()>;
+    fn accept(&self, flags: FileFlags) -> Result<(Self, Option<SockAddr>)>
```

Comment:
Why returning `Option<SockAddr>`? I think an address can always be returned from `accept`.

Reply:
It happens when no address is returned. The POSIX API returns zero-length.